components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -568,69 +562,36 @@ class NewTabPage extends React.Component<Props, State> {
-    })
-  }
-
-  getCryptoDotComChartData = (asset: string) => {
-    return new Promise((resolve: Function) => {
-      chrome.cryptoDotCom.getChartData(`${asset}_USDT`, (resp: any) => {
-        resolve({ [asset]: resp })
-      })
-    })
-  }
-
-  getCryptoDotComSupportedPairs = () => {
-    return new Promise((resolve: Function) => {
-      chrome.cryptoDotCom.getSupportedPairs((resp: any) => {
-        resolve(resp)
-      })
-    })
-  }
-
-  fetchCryptoDotComTickerPrices = async (assets: string[]) => {
-    const assetReqs = assets.map(asset => this.getCryptoDotComTickerInfo(asset))
-    const assetResps = await Promise.all(assetReqs).then((resps: object[]) => resps)
-    const assetPrices = assetResps.reduce((all, current) => ({ ...current, ...all }), {})
-    this.props.actions.setCryptoDotComTickerPrices(assetPrices)
-  }
-
-  fetchCryptoDotComCharts = async (assets: string[]) => {
-    const chartReqs = assets.map(asset => this.getCryptoDotComChartData(asset))
-    const chartResps = await Promise.all(chartReqs).then((resps: object[]) => resps)
-    const charts = chartResps.reduce((all, current) => ({ ...current, ...all }), {})
-    this.props.actions.setCryptoDotComCharts(charts)
-  }
-
-  fetchCryptoDotComLosersGainers = async () => {
-    const losersGainers = await this.getCryptoDotComAssetRankings().then((resp: any) => resp)
-    this.props.actions.setCryptoDotComLosersGainers(losersGainers)
+  onCryptoDotComMarketsRequested = async (assets: string[]) => {
+    const [tickerPrices, losersGainers] = await Promise.all([
+      fetchCryptoDotComTickerPrices(assets),
+      fetchCryptoDotComLosersGainers()
+    ])
+    this.props.actions.setCryptoDotComMarketsRequested(tickerPrices, losersGainers)
   }
 
-  fetchCryptoDotComSupportedPairs = async () => {
-    const pairs = await this.getCryptoDotComSupportedPairs().then((resp: any) => resp)
-    this.props.actions.setCryptoDotComSupportedPairs(pairs)
+  onCryptoDotComAssetData = async (assets: string[]) => {
+    const [charts, pairs] = await Promise.all([
+      fetchCryptoDotComCharts(assets),
+      fetchCryptoDotComSupportedPairs()
+    ])
+    this.props.actions.setCryptoDotComAssetData(charts, pairs)
   }
 
   cryptoDotComUpdateActions = async () => {
     const assets = Object.keys(this.props.newTabData.cryptoDotComState.tickerPrices)
-    return Promise.all([
-      this.fetchCryptoDotComTickerPrices(assets),
-      this.fetchCryptoDotComLosersGainers(),
-      this.fetchCryptoDotComCharts(assets)
+    const [tickerPrices, losersGainers, charts] = await Promise.all([
+      fetchCryptoDotComTickerPrices(assets),
+      fetchCryptoDotComLosersGainers(),
+      fetchCryptoDotComCharts(assets)
     ])
+    this.props.actions.onCryptoDotComRefreshData(tickerPrices, losersGainers, charts)
+  }
+
+  onBtcPriceOptIn = async () => {
+    this.props.actions.onBtcPriceOptIn()
+    this.props.actions.onCryptoDotComInteraction()
+    await this.onCryptoDotComMarketsRequested(['BTC'])
```

Comment:
**follow-up:** move all this logic to an async redux handler, (and don't dispatch separate `onCryptoDotComInteraction` events, preferring to call the neccessary API when the appropriate redux event is dispatched - names of redux events should correspond to user actions and API data events anyway).

Reply:
Can you point me to an example in the repo of an async redux handler? When I attempted to place the logic in an async action, I received errors.